src/evict/evict_page.c

```diff
@@ -873,9 +875,16 @@ __evict_reconcile(WT_SESSION_IMPL *session, WT_REF *ref, uint32_t evict_flags)
          * outside world.
          */
         __wt_txn_bump_snapshot(session);
-    else if (use_snapshot_for_app_thread)
+    else if (use_snapshot_for_app_thread) {
+        /*
+         * Application threads entering into eviction saves the existing snapshots and refresh to
+         * acquire a new snapshot to evict the latest data, once the application threads are done
+         * with eviction then the snapshots are switched back to its original snapshots.
+         */
+        ret = __wt_txn_snapshot_save_and_refresh(session);
```

Comment:
I would prefer to pass in a bool flag to this function to indicate whether the snapshot is refreshed or not. If ret is not 0, we should handle the error instead of continue.

Reply:
I didn't check the entire PR, just by reading the comment, how about the function itself take care of any failures internally and on success means that it has successfully switched the snapshot. Upon any further failures, we must switch back to the original snapshot.

Reply:
Fixed in a5ecfa2de7d6c80534029e741e036ec37eb4e955